test/data/multiline.csv

```diff
@@ -0,0 +1,32 @@
+// this file should be loaded with quoting enabled
+// and it should load without problems by the TextLoader
+id,description,animal
+// this is a comment that will be ignored
+// this is a comment with "quotes" that will also be ignored
+// this is a comment with a "quote without close quote that will also be ignored
+10,this is a description,dog
+11,"this is a quoted description",cat
+12,"this is a multiline
+quoted description", bird
+13,"this has one""doublequote which should be escaped as a single quote",dog
+14,"this has ""doublequotes"" inside of it",cat
+15, "this is a multiline
+quoted description with
+
+""doublequotes"" and
+
+empty new lines and
+
+
+
+escaped quotes inside of ""it""
+
+//and this comment with // shouldn't be ignored
+since it is part of the ""multiline""",bird
+// this line should be ignored, and the next line is empty:
+
+16, here is text after the empty line, dog
+17, this is a line with an empty animal,""
+"", this is a line with an empty id, bird
+19,"",dog
+20,this is the last row description,cat
```

Comment:
_(pinned to a line of code to allow for threaded conversations)_

We should check the behavior of the `TextSaver` to see if it saves the newlines as expected. We should be able to read a file w/ the `TextLoader` and write it back to disk using the `TextSaver` and have it be identical.

Reply:
TextSaver didn't support fields with new lines correctly, and I've fixed this in a more recent iteration. Also added a couple of tests on it.

---
In reply to: [425779339](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425779339) [](ancestors = 425779339)

Reply:
Notice that in general it's not necessarily true that the original text file and the one created by TextSaver are going to be identical. Some information is lost when using TextLoader: on one hand options such as the delimitator that was used in the file (which could be different when using TextSaver's options), and also `//comments` which are removed from the file, as well as`  ,"",` missing values which might have been replaced by something else.

So what I actually tested is the more general case where both following dataviews should be the same (instead of comparing the original files, what is compared are the dataviews):
1. Loading a dataview from text
2. Loading a dataview from text, saving it back to text with TextSaver with whatever format, and loading it back into a new dataview.
Both dataviews should be the same.

---
In reply to: [426977715](https://github.com/dotnet/machinelearning/pull/5125#discussion_r426977715) [](ancestors = 426977715,425779339)